pkg/kubelet/prober/prober_manager.go

```diff
@@ -241,6 +241,9 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *v1.PodStatus) {
 			ready = false
 		} else if result, ok := m.readinessManager.Get(kubecontainer.ParseContainerID(c.ContainerID)); ok {
 			ready = result == results.Success
+		} else if c.Ready {
+			// keep the readiness until we get a new result from readinessManager
```

Comment:
It might be better if adding what cases you are dealing with.

Reply:
Ok, I'll add the issue number in the comment